x/oracle/keeper/aggregator/aggregator.go

```diff
@@ -48,6 +48,34 @@ type aggregator struct {
 	dsPrices map[uint64]string
 }
 
+func (agg *aggregator) copy4CheckTx() *aggregator {
+	ret := &aggregator{
+		finalPrice:  big.NewInt(agg.finalPrice.Int64()),
+		reportPower: big.NewInt(agg.reportPower.Int64()),
+		totalPower:  big.NewInt(agg.totalPower.Int64()),
+
+		reports:  make([]*reportPrice, 0, len(agg.reports)),
+		dsPrices: agg.dsPrices,
```

Comment:
`**dsPrices: agg.dsPrices,**`this line of code is making a Shallow Copy, while This follwoing code is a manual attempt to perform Deep Copy.
`	for k, v := range agg.dsPrices {
		ret.dsPrices[k] = v
	}`

It is not recommended to use both pieces of code at the same time. First decide whether you need shallow or deep copies, and then choose your implementation accordingly

Reply:
`ds` is for deterministic-source, corresponding to `ns`: nondeteministic-source
`detID` for deterministicID